### PR TITLE
FO-9322 - cert-fix checking the status of default ingress and kubeapi-server after patching

### PR DESCRIPTION
**Pull request by erzhan46**

### CODE REVIEW

File: cert-tool/cert-tool.sh [hunk 1]
```diff
@@ -449,6 +457,61 @@ function load_certs() {
 }
 
 
+# DESC: Load current state
+# ARGS: none
+# OUTS: None if successful, Error text otherwise
+function load_current_state() {
+    INGRESS_GENERATION=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.observedGeneration')
```
Comment (vladimir-babichev) [inline cert-tool/cert-tool.sh]: Just a note, wouldn't it be faster to run only one call against ARO API and then parse response with `jq`. Ie:
```bash
ROUTER_RESOURCE=$(${oc_cmd} get deployment router-default -n openshift-ingress)
INGRESS_GENERATION=$(echo ${ROUTER_RESOURCE} | jq -r '.status.observedGeneration')
...
```

File: cert-tool/cert-tool.sh [hunk 2]
```diff
@@ -449,6 +457,61 @@ function load_certs() {
...
+        if [[ ${C_INGRESS_GENERATION} -eq ${NEXT_INGRESS_GENERATION} && ${INGRESS_REPLICAS} -eq ${C_INGRESS_REPLICAS} && ${INGRESS_AVAILABLEREPLICAS} -eq ${C_INGRESS_AVAILABLEREPLICAS} && ${INGRESS_READYREPLICAS} -eq ${C_INGRESS_READYREPLICAS} && ${INGRESS_UPDATEDREPLICAS} -eq ${C_INGRESS_UPDATEDREPLICAS} ]]; then
+            INGRESS_COMPLETE=true
+        fi
+        # Checking kube-apiserver operator status
+        read -r C_API_NAME C_API_VERSION C_API_AVAILABLE C_API_PROGRESSING C_API_DEGRADED C_API_SINCE <<< "$(${oc_cmd} get clusteroperators kube-apiserver | grep -v 'NAME')"
+        if [[ ${C_API_AVAILABLE} == "True" && ${C_API_PROGRESSING} == "False" && ${C_API_DEGRADED} == "False" ]]; then
+            API_COMPLETE=true
+        fi
+
+        sleep ${SLEEP}
```
Comment (vladimir-babichev) [inline cert-tool/cert-tool.sh:513]: Is there a need for this sleep? From what I can see, this function is called only once.

Comment (erzhan46) [reply cert-tool/cert-tool.sh:513]: There is a for loop - which will keep asking for status every 60 seconds (total 16 iterations)
Mostly for kubeapi-server to complete transition

File: cert-tool/cert-tool.sh (hunk 1 above)
Comment (erzhan46) [reply cert-tool/cert-tool.sh]: Done

**Review by vladimir-babichev — APPROVED** [review]

**Pull request merged**